OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -204,10 +217,33 @@ void DeployMcv(IBot bot, Actor mcv, bool move)
 				return null;
 			};
 
-			var baseCenter = GetRandomBaseCenter();
+			var baseCenter = GetRandomBaseCenter(distanceToBaseIsImportant);
+
+			var bc = findPos(baseCenter, baseCenter, Info.MinBaseRadius, Info.MaxBaseRadius);
+
+			if (!bc.HasValue)
+				return null;
+
+			baseCenter = bc.Value;
+
+			var wPos = world.Map.CenterOfCell(baseCenter);
+			var newBaseRadius = new WDist(Info.MaxBaseRadius * 1024);
+
+			var actors = world.FindActorsInCircle(wPos, newBaseRadius)
+				.Where(a => !a.Disposed);
+
+			var enemies = actors.Where(a => player.Stances[a.Owner] == Stance.Enemy);
+
+			if (enemies.Count() > 0)
+				return null;
+
+			var self = actors.Where(a => a.Owner == player
+					&& (Info.McvTypes.Contains(a.Info.Name) || (Info.ConstructionYardTypes.Contains(a.Info.Name) && a.Info.HasTraitInfo<BuildingInfo>())));
```

Comment:
Why does this check for `BuildingInfo` alongside of checking `ConstructionYardTypes`?

Reply:
I think that is possibly an oversight and something I forgot to remove.